crates/payload/primitives/src/traits.rs

```diff
@@ -163,3 +163,12 @@ impl PayloadAttributes for OptimismPayloadAttributes {
         Ok(())
     }
 }
+
+/// A builder that can return the current payload attribute.
+pub trait PayloadAttributesBuilder: std::fmt::Debug + Send + Sync + 'static {
+    /// The payload attributes type returned by the builder.
+    type PayloadAttributes: PayloadAttributes;
+
+    /// Return a new payload attribute from the builder.
+    fn build(&self) -> Self::PayloadAttributes;
```

Comment:
I wanna make this return a result, in case the impl needs to do some io (e.g. fetch current head, etc)